diff --git a/node_modules/react-native-crossfade-image/src/index.tsx b/node_modules/react-native-crossfade-image/src/index.tsx
index 2089d6f..ce3b645 100644
--- a/node_modules/react-native-crossfade-image/src/index.tsx
+++ b/node_modules/react-native-crossfade-image/src/index.tsx
@@ -51,15 +51,15 @@ export const CrossfadeImage: React.FC<CrossfadeImageProps> = ({
   }, [opacity, newSource, duration, easing]);
 
   const handleUpdate = useCallback(() => {
-    setNewSource(undefined);
     opacity.setValue(0);
+    setNewSource(undefined);
   }, [opacity]);
 
   return (
     <View style={[styles.root, style]}>
-      <Image
+      <Animated.Image
         {...props}
-        style={styles.image}
+        style={[styles.image, { opacity: opacity.interpolate({ inputRange: [0, 1], outputRange: [1, 0] }) }]}
         source={oldSource}
         fadeDuration={0}
         onLoad={handleUpdate}
